src/pcapi/core/offers/api.py

```diff
@@ -478,3 +481,35 @@ def get_expense_domains(offer: Offer) -> List[ExpenseDomain]:
             domains.add(ExpenseDomain.PHYSICAL.value)
 
     return list(domains)
+
+
+def add_criteria_to_offers(criteria: List[Criterion], isbn: str) -> bool:
+    isbn = isbn.replace("-", "").replace(" ", "")
+
+    offer_ids_tuples = (
+        Offer.query.filter(Offer.extraData["isbn"].astext == isbn)
+        .filter(Offer.isActive.is_(True))
+        .with_entities(Offer.id)
+        .all()
+    )
+    offer_ids = [offer_id[0] for offer_id in offer_ids_tuples]
+
+    if not offer_ids:
+        return False
+
+    for criterion in criteria:
+        logger.info("Adding criterion %s to %d offers", criterion, len(offer_ids))
+
+        offer_criteria = []
+        for offer_id in offer_ids:
+            offer_criteria.append(OfferCriterion(offerId=offer_id, criterionId=criterion.id))
+
+            db.session.bulk_save_objects(offer_criteria)
+            db.session.commit()
+            offer_criteria = []
+
+        if feature_queries.is_active(FeatureToggle.SYNCHRONIZE_ALGOLIA):
+            for offer_id in offer_ids:
+                redis.add_offer_id(client=app.redis_client, offer_id=offer_id)
```

Comment:
On ne voudrait pas plutôt `bulk_save` tous les OfferCriterion de toutes les offres en une fois ? Je pense que généralement, on va a ajouter 1 tag à N offres. Du coup, ici, ça ferait 1 requête + 1 commit par offre. On perd un peu l'intérêt du bulk, du coup.

Je pensais plutôt à :

```suggestion
    offer_criteteria = []
    for criterion in criteria:
        logger.info("Adding criterion %s to %d offers", criterion, len(offer_ids))
        offer_criteria.extend(OfferCriterion(offerId=offer_id, criterionId=criterion.id) for offer_id in offer_ids)

    db.session.bulk_save_objects(offer_criteria)
    db.session.commit()

    if feature_queries.is_active(FeatureToggle.SYNCHRONIZE_ALGOLIA):
        for offer_id in offer_ids:
            redis.add_offer_id(client=app.redis_client, offer_id=offer_id)
```

au passage, ça évite d'ajoute X fois la même offre dans le Redis pour réindexation quand on ajoute X tags à une offre.